Foundation/NSData.swift

```diff
@@ -414,6 +421,39 @@ open class NSData : NSObject, NSCopying, NSMutableCopying, NSSecureCoding {
             free(buffer)
         }
     }
+
+#if os(Linux)
+    internal static func readZeroSizeFile(_ fd: Int32) throws -> NSDataReadResult {
+        let blockSize = 1024 * 1024 // 1MB
+        var data: UnsafeMutableRawPointer? = nil
+        var bytesRead = 0
+        var amt = 0
+
+        repeat {
+            data = realloc(data, bytesRead + blockSize)
```

Comment:
I was duplicating the usage of `malloc()` in the `readBytesFromFileWithExtendedAttributes` function above. Generally the pattern in Foundation seems to be not checking the return value of `malloc()/realloc()` and letting it eventually abort on a NULL deref. The same is true if `bytesRead` exceeds `Int.max`